src/packet/signature.js

```diff
@@ -672,7 +671,8 @@ class SignaturePacket {
    * @param {String|Object} data data which on the signature applies
    * @param {Boolean} detached (optional) whether to verify a detached signature
    * @param {Boolean} streaming (optional) whether to process data as a stream
-   * @returns {Promise<Boolean>} True if message is verified, else false.
+   * @return true if the signature is valid
```

Comment:
I wanted to also change this one to always return nothing, but it turns out we check its returned value in several tests, as well as in `getLatestValidSignature` -- I would still change it, @twiss what do you think?

Reply:
Yeah, for this function the returned boolean used to actually be significant. I think there's a few other places too if you do a global search for `verified ||`. Which - maybe we should fold that into this function - and also the signature expiry checking. E.g. [this](https://github.com/openpgpjs/openpgpjs/blob/2eab8a1ebcd84f913f948c6da845cd66020ec170/src%2Fkey%2Fhelper.js#L252-L269):
```js
    try {
      if (
        (!signature || revocationSignature.issuerKeyId.equals(signature.issuerKeyId)) &&
        !(config.revocations_expire && revocationSignature.isExpired(normDate)) &&
        (revocationSignature.verified || await revocationSignature.verify(key, signatureType, dataToVerify))
      ) {
        revocationKeyIds.push(revocationSignature.issuerKeyId);
      }
    } catch (e) {}
```
should probably just be something like
```js
    if (!signature || revocationSignature.issuerKeyId.equals(signature.issuerKeyId)) {
      try {
        await revocationSignature.verify(key, signatureType, dataToVerify));
        revocationKeyIds.push(revocationSignature.issuerKeyId);
      } catch (e) {}
    }
```
(but with comments intact). Though that probably shouldn't happen in this PR. But anyway yes I think you can drop the return value and change all the calling locations ^.^

Reply:
You mean we should ignore the "cached" `.verified` value, and re-verify every time?

Reply:
I meant the former. I guess the most concrete case is if you call `.verify` twice with different data, or different keys. Which.. for (detached) message signatures is actually kinda plausible, so yeah maybe we should keep it outside indeed. Though - for this to affect the result you'd have to first verify with the _correct_ message & key, and _then_ with a wrong one. I don't really see a concrete use-case for that, but I guess it's better to be correct.